src/analysis/retail/priest/holy/modules/talents/BottomRow/Lightweaver.tsx

```diff
@@ -30,44 +44,154 @@ class Lightweaver extends Analyzer {
       this.active = false;
       return;
     }
-    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL), this.onHeal);
     this.addEventListener(
-      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.LIGHTWEAVER_TALENT_BUFF),
-      this.onBuff,
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL),
+      this.onHealCast,
     );
     this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.LIGHTWEAVER_TALENT_BUFF),
-      this.onBuff,
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.FLASH_HEAL),
+      this.onFlashHealCast,
     );
   }
 
-  onHeal(Event: HealEvent) {
-    if (this.currentStacks > 0) {
-      this.healingDoneFromTalent += calculateEffectiveHealing(
-        Event,
-        HEALING_BONUS_PER_STACK * this.currentStacks,
-      );
-      this.overhealingDoneFromTalent += calculateOverhealing(
-        Event,
-        HEALING_BONUS_PER_STACK * this.currentStacks,
-      );
-
-      this.currentStacks = 0;
-      this.timesUsedBuffed += 1;
-    } else {
-      this.timesUsedUnbuffed += 1;
+  isHighOverheal(event: HealEvent) {
+    const rawHealing = (event.amount || 0) + (event.overheal || 0) + (event.absorbed || 0);
+    if (rawHealing === 0) {
+      return false;
+    }
+    return (event.overheal || 0) / rawHealing >= OVERHEAL_THRESHOLD;
+  }
+
+  onHealCast(event: CastEvent) {
+    // linked heal event exists
+    const healEvent = getHeal(event);
+    if (healEvent) {
+      this.totalHealCasts += 1;
+
+      if (isCastBuffedByLightweaver(event)) {
+        // calculate effective healing from bonus
+        this.healingDoneFromTalent += calculateEffectiveHealing(healEvent, HEALING_BONUS);
+        this.overhealingDoneFromTalent = calculateOverhealing(healEvent, HEALING_BONUS);
```

Comment:
this looks wrong, unless the stacks of the buff act more like charges (so a cast consumes one stack for 15% bonus healing)

Reply:
only one stack of lightweaver is consumed per cast of heal, so yes it is more like charges. the live implementation of this statistic is incorrectly calculating this since it is treating it like both charges are consumed for 30%. that is not how it works in game.